src/meta/api/src/schema_api_impl.rs

```diff
@@ -2022,23 +2267,28 @@ async fn gc_dropped_table(
                 // condition: database exists
                 txn_cond_seq(tenant_dbname, Eq, db_id_seq),
             ];
-            // remove table id keys not changed
-            for key in remove_table_keys.iter() {
-                condition.push(txn_cond_seq(&key.0, Eq, key.1));
-            }
             let mut if_then = vec![
                 // save new table id list
                 txn_op_put(&dbid_tbname_idlist, serialize_struct(&new_tb_id_list)?),
             ];
-            // remove out of time table meta
+
             for key in remove_table_keys.iter() {
+                // remove table id keys not changed
+                condition.push(txn_cond_seq(&key.0, Eq, key.1));
+
+                // remove out of time table meta
                 if_then.push(txn_op_del(&key.0));
+
+                // remove stage file info of the table
+                remove_table_copied_files(kv_api, key.0.table_id, &mut condition, &mut if_then)
+                    .await?;
```

Comment:
It looks like it does not assert the seq of the TableCopiedFileLock.

BTW consider moving asserting seq of TableCopiedFileLock into `remove_table_copied_files()`?